framework/include/utils/PiecewiseByBlockLambdaFunctor.h

```diff
@@ -261,7 +275,15 @@ PiecewiseByBlockLambdaFunctor<T>::evaluate(const Moose::FaceArg & face,
 {
   using namespace Moose::FV;
   mooseAssert(state == 0, "Only current time state supported.");
-  return interpolate(*this, face);
+
+  if (isInternalFace(*face.fi))
+  {
+    return interpolate(*this, face);
+  }
+
+  Moose::SingleSidedFaceArg ssfa = {
+      face.fi, face.limiter_type, face.elem_is_upwind, face.correct_skewness, face.elem_sub_id};
```

Comment:
Instead of using `face.elem_sub_id` always, we should base this off of what side of the interface this functor is defined on. It may only be on the neighbor side instead of the elem side

Reply:
Added a conditional subdomain ID based on the `hasBlock(elemID)`.